### PR TITLE
remove warnings

### PR DESCRIPTION
**Pull request by matsl**
# What

- Remove unused warning, comment out all occurences of new-buf-flag
- Add defvar and declare-function to silence warnings

# Why

We should minimize the number of warnings.

# Note

I'm keeping the new-buf-flag but commenting it put everywhere to be in
company with the comment later in the code. I would opt for removing
all of these comments.

### CODE REVIEW

File: hyrolo.el [hunk 1]
```diff
@@ -1973,7 +1973,8 @@ Return number of matching entries found."
   ;; Save pattern as last rolo search expression.
   (setq hyrolo-match-regexp pattern)
   ;;
-  (let ((new-buf-flag) (actual-buf)
+  (let (;;(new-buf-flag)
```
Comment (matsl) [inline hyrolo.el:1976]: I propose we remove this altogether with the comment on line 2002. But the suggested change is a middle ground if the comment should stay.

File: hui-treemacs.el [hunk 2]
```diff
@@ -45,6 +45,8 @@
 (declare-function treemacs-node-buffer-and-position "ext:treemacs-mouse-interface")
 (declare-function treemacs-quit "ext:treemacs-interface")
 (declare-function treemacs-toggle-node "ext:treemacs-interface")
+(declare-function treemacs-add-and-display-current-project-exclusively "ext:treemacs")
```
Comment (rswgnu) [inline hui-treemacs.el:48]: Alphabetize these declarations.

File: hywiki.el [hunk 3]
```diff
@@ -139,6 +139,7 @@
 (defvar org-agenda-buffer-tmp-name)  ;; "org-agenda.el"
 (defvar org-export-with-broken-links);; "ox.el"
 (defvar org-publish-project-alist)   ;; "ox-publish.el"
+(defvar action-key-modeline-buffer-id-function)  ;; "hui-mouse"
```
Comment (rswgnu) [inline hywiki.el:142]: Alphabetize.

File: hyrolo.el (hunk 1 above)
Comment (rswgnu) [reply hyrolo.el:1976]: So you are saying it is not used anywhere?  If so, just remove it.  If there are no other references to it anywhere in the code and the return value of its so setq is not used then it is just a leftover from some prior version and you should remove it.

**Review by rswgnu — CHANGES_REQUESTED** [review]

**Review by rswgnu — APPROVED** [review]

**Pull request merged**